provider/gce/environ_network.go

```diff
@@ -155,46 +162,64 @@ func (e *environ) NetworkInterfaces(ctx context.ProviderCallContext, instId inst
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	googleInst := envInst.base
-	ifaces := googleInst.NetworkInterfaces()
 
-	var subnetURLs []string
-	for _, iface := range ifaces {
-		if iface.Subnetwork != "" {
-			subnetURLs = append(subnetURLs, iface.Subnetwork)
+	var (
+		infos      = make([][]network.InterfaceInfo, len(ids))
+		subnetURLs []string
+	)
+
+	for idx, inst := range insts {
+		if inst == nil {
+			continue // no instance with this ID known by provider
+		}
+
+		envInst, ok := inst.(*environInstance)
+		if !ok { // This shouldn't happen.
+			return nil, errors.Errorf("couldn't extract google instance for %q", ids[idx])
+		}
+
+		subnetURLs = subnetURLs[:0]
+		ifaces := envInst.base.NetworkInterfaces()
+		for _, iface := range ifaces {
+			if iface.Subnetwork != "" {
+				subnetURLs = append(subnetURLs, iface.Subnetwork)
+			}
 		}
-	}
-	subnets, err := e.subnetsByURL(ctx, subnetURLs, networks, zones)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	// We know there'll be a subnet for each url requested, otherwise
-	// there would have been an error.
 
-	var results []network.InterfaceInfo
-	for i, iface := range ifaces {
-		details, err := findNetworkDetails(iface, subnets, networks)
+		subnets, err := e.subnetsByURL(ctx, subnetURLs, networks, zones)
```

Comment:
Is it possible to loop over the instances separately to collect `subnetURLs` for a single call to this method?
Looks like an efficiency gain.

Reply:
Good idea. I will push an update to address this!